Packet++/header/NflogLayer.h

```diff
@@ -103,6 +103,18 @@ namespace pcpp
 		 */
 		size_t getTotalSize() const { return m_Data->recordLen; }
 
+		/**
+		 * @return The total size of the TLV record (in bytes) aligned by 4
+		 */
+		size_t getTotalSizeAligned() const
```

Comment:
Why do we need an additional method `getTotalSizeAligned()` instead of modifying `getTotalSize()` in `NflogTlv` to be aligned by 4?

In the packet example [`nflogPacket.pcap`](https://github.com/seladb/PcapPlusPlus/blob/master/Tests/Packet%2B%2BTest/PacketExamples/nflogPacket.pcap) we have such TLV record which length is 5 but the actual length is 8:

![image](https://github.com/seladb/PcapPlusPlus/assets/9059541/83033c2f-cabe-47b7-b47f-25cfdcd10772)

Reply:
I see your point. However in PcapPlusPlus we usually don't have "gaps" in the data, meaning a TLV record has to start immediately after the previous one.

I looked at the code again, `NflogTlv` doesn't inherit from `TLVRecord` so it doesn't inherit its equality operator. We can implement an equality operator for `NflogTlv` that compares just the "non garbage" part instead of the whole TLV buffer. Also, I don't think `TLVRecordReader` actually uses the equality operator so it's kind of optional...

NFLOG is quite unique in its approach of allowing TLV records to be of unaligned size, but require that the next TLV record starts at an aligned position, so I think it's better to make a "local" fix that is only relevant for NFLOG.

Please let me know what you think.